src/System.Runtime.Extensions/src/System/Environment.Win32.cs

```diff
@@ -341,13 +342,17 @@ public static string UserName
                 }
 
                 return string.Empty;
+#else
+                return "Windows User";
+#endif
```

Comment:
Understand, but I find it more confusing when we use both approaches together.  We've been doing a (relatively) good job of avoiding ifdef hell, and we've done that with partial classes, so that's the general plan we're on.

Why do we have both Environment.Windows.cs and Environment.Win32.cs?  From a quick look at the project file, it looks like they're both included at the same time.  In which case we wouldn't need three different Windows files, we could just have two, e.g. the .Windows.cs could have a UserName property implemented something like:
```C#
public string UserName
{
    get
    {
        string username = "Windows User";
        GetUserName(ref username);
        return username;
    }
}

partial void GetUserName(ref string username);
```
and then a .Win32.cs file could have:
```
partial void GetUserName(ref string username)
{
    ... // implementation that uses Win32 APIs and sets username to the result
}
```
Then we wouldn't need a UAP specific file.

Or something like that.